src/applications/vaos/reducers/newAppointment.js

```diff
@@ -591,6 +591,8 @@ export default function formReducer(state = initialState, action) {
           additionalInfoTitle,
           reasonSchema,
         );
+      } else {
+        delete formData.reasonForAppointment;
```

Comment:
In the rare care where a user gets past the reason page on a VA direct schedule or request, and then goes back in the flow and switches to community care, this clears our the reasonForAppointment.